modules/standard/BigInteger.chpl

```diff
@@ -516,56 +516,32 @@ module BigInteger {
       return (dbl: real, exp.safeCast(uint(32)));
     }
 
-    proc get_str(base: int = 10) : string {
+    // private method
+    @chpldoc.nodoc
+    proc getStr(base: int = 10): string {
       const base_ = base.safeCast(c_int);
       var   ret: string;
 
-      if compiledForSingleLocale() {
-        var tmpvar = chpl_gmp_mpz_get_str(base_, this.mpz);
-
-        try! {
-          ret = string.createAdoptingBuffer(tmpvar);
-        }
-
-      } else if this.localeId == chpl_nodeID {
+      if compiledForSingleLocale() ||  this.localeId == chpl_nodeID {
```

Comment:
I think the `compiledForSingleLocale()` is a `param`, so that is why these were separated out into different branches. It's difficult to say if that is important, but that's why this was here originally. Now, in single locale configurations, we will have the branch and construction of the locale object since we can't fold that out, which can cause some overhead.

Reply:
I did test compiling this in a single locale configuration and it does not generate the extra check (looking at AST for 'codegen' pass). It is short enough code that I can also just put it back